src/west/util.py

```diff
@@ -20,3 +21,26 @@ def wrap(text, indent):
     '''Convenience routine for wrapping text to a consistent indent.'''
     return textwrap.wrap(text, initial_indent=indent,
                          subsequent_indent=indent)
+
+
+class WestNotFound(RuntimeError):
+    '''Neither the current directory nor any parent has a West installation.'''
+
+
+def west_topdir():
+    '''
+    Returns the absolute path of the first directory containing a .west/
+    directory, searching the current directory and its parents.
+
+    Raises WestNotFound if no .west/ directory is found.
+    '''
+    search_dir = os.getcwd()
+
+    # While the directory is not the root directory...
+    while search_dir != os.path.dirname(search_dir):
```

Comment:
Noticed a subtle bug here. Won't find a `.west/` in the root directory.

Guessing it's no panic though. :P

Reply:
Heh, whoops :), nice catch!